x/programs/examples/imports/pstate/pstate.go

```diff
@@ -217,9 +217,45 @@ func (i *Import) deleteFn(caller *program.Caller, memOffset int32, size int32) (
 	}
 
 	k := storage.ProgramPrefixKey(args.ProgramID[:], args.Key)
+	bytes, err = i.mu.GetValue(context.Background(), k)
+	if err != nil {
+		if errors.Is(err, database.ErrNotFound) {
+			val, err := program.WriteBytes(memory, []byte{0})
```

Comment:
What is the reasoning behind writing a single `0` byte to memory, is that to get a pointer ?

Reply:
Good question, I need to add a comment

Reply:
Added, let me know if that's good enough or if you think I need to add more detail

Reply:
It all makes sense, thanks !